File: Week_01/G20190343020229/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        v_map = {}
        for idx, i in enumerate(nums):
            # print(v_map)
            if i in v_map:
                return [v_map[i], idx]
            else:
                left = target - i
                v_map[left] = idx
        return []

# @lc code=end

File: Week_02/G20190343020149/1.两数之和.go
package leetcode

/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	numCache := map[int]int{}
	for i, num := range nums {
		if val, ok := numCache[target-num]; ok {
			return []int{val, i}
		}
		numCache[num] = i
	}
	return []int{}
}

// @lc code=end
